### ISSUE TITLE
udev: the broken symbolic links left in system

### ISSUE THREAD

**Issue by gaoyi1988** [systemd/systemd#19946]
**systemd version the issue has been seen with**
 > 246

**Expected behaviour you didn't see**
 > There is no broken symbolic links in /dev

**Unexpected behaviour you saw**
 > There are some broken symbolic links in /dev

**Steps to reproduce the problem**
 > 1. parted -s /dev/sdb mkpart logic (n)M (n+2)M，create 50 partitions for /dev/sdb
> 2. parted -s /dev/sdb rm $i,  delete partitions created in step 1
> 3. repeat step 1 and 2

**Additional program output to the terminal or log subsystem illustrating the issue**

```text
Jun 12 21:54:28 localhost.localdomain systemd-udevd[961677]: sdb37: Processing device (SEQNUM=5318522, ACTION=change)
Jun 12 21:54:28 localhost.localdomain systemd-udevd[961677]: sdb37: Found node '/dev/sdb37' with non-matching devnum b259:21, skip handling
Jun 12 21:54:28 localhost.localdomain systemd-udevd[961677]: sdb37: Failed to process device, ignoring: File exists
```

**Comment by gaoyi1988** [comment]
I add some additional logs, and find as follow:
1. udev_event_execute_rules() -> device_clone_with_db(), when clone with db, the devnum for /dev/sdb37 is b259:9
2. For other locations in udev_event_execute_rules(), the devnum is b259:21

**Comment by gaoyi1988** [comment]
There seems to have another problem, that when I change the log_level to debug in /etc/udev/udev.conf, the debug logs won't print when "File exists" error occured.
The follow logs recorded in journal just when I change the code to "static int log_max_level[] = {LOG_DEBUG, LOG_DEBUG};" in log.c
```
Jun 12 21:54:28 localhost.localdomain systemd-udevd[961677]: sdb37: Processing device (SEQNUM=5318522, ACTION=change)
Jun 12 21:54:28 localhost.localdomain systemd-udevd[961677]: sdb37: Found node '/dev/sdb37' with non-matching devnum b259:21, skip handling
```

**Comment by yuwata** [comment]
Hm, the patches for the previous similar issue are applied?

**Comment by yuwata** [comment]
Maybe, the following happens. If `change`, `remove`, uevents for a device node are triggered within a short time period, then
1. the `change` event fails at somewhere as the device is already removed, devlinks are not saved in udev database, and
2. the `remove` event does not remove the existing symlinks.

**Comment by yuwata** [comment]
What kind of broken symlinks exist in /dev?

**Comment by gaoyi1988** [comment]
> Hm, all the patches in PR #19796 for the previous similar issue are applied?



> What kind of broken symlinks exist in /dev?

Like the follows:
```
lrwxrwxrwx. 1 root root 11 Jun 12 21:54 scsi-0QEMU_QEMU_HARDDISK_drive-scsi0-0-0-4-part25 -> ../../sdb37
```

**Comment by gaoyi1988** [comment]
> Maybe, the following happens. If `change`, `remove`, uevents for a device node are triggered within a short time period, then
> 
> 1. the `change` event fails at somewhere as the device is already removed, devlinks are not saved in udev database, and
> 2. the `remove` event does not remove the existing symlinks.

@yuwata Do you mean the issue https://github.com/systemd/systemd/issues/19788, this patches are applied, but not solve the problem.

I think this is more like the devices competition conflict due to the devnum changes when partition creation/deletion.

**Comment by yuwata** [comment]
Yeah, you are right. The patch for the previous issue is #19815.

**Comment by yuwata** [comment]
Fix is waiting in #20603.
